remote-access/remote_connect.py

```diff
@@ -0,0 +1,77 @@
+import logging
+from c8ydp.device_proxy import DeviceProxy, WebSocketFailureException
+from threading import Thread
+import threading
+from c8yMQTT import C8yMQTT
+import concurrent.futures
+
+logging.basicConfig(level=logging.INFO,format='%(asctime)s %(name)s %(message)s')
+logger = logging.getLogger(__name__)
+
+
+def setCommandExecuting(command):
+    logger.info('Setting command: '+ command + ' to executing')
+    c8y.publish('c8y/s/us','501,'+command)
+
+def setCommandSuccessfull(command):
+    logger.info('Setting command: '+ command + ' to successful')
+    c8y.publish('c8y/s/us','503,'+command)
+
+def setCommandFailed(command,errorMessage):
+    logger.info('Setting command: '+ command + ' to failed cause: ' +errorMessage)
+    c8y.publish('c8y/s/us','502,'+command+','+errorMessage)
+
+def on_message(client, obj, msg):
+    message = msg.payload.decode('utf-8')
+    logger.debug("Message Received: " + msg.topic + " " + str(msg.qos) + " " + message)
+    if message.startswith('71'):
+        fields = message.split(",")
+        c8y.token = fields[1]
+        logger.info('New JWT Token received')
+    if message.startswith('530'):
+        fields = message.split(",")
+        tcp_host = fields[2]
+        tcp_port = int(fields[3])
+        connection_key = fields[4]
+        c8y.logger.info('Received Remote Connect.')
+        setCommandExecuting('c8y_RemoteAccessConnect') 
+        with concurrent.futures.ThreadPoolExecutor() as executor:
+            future = executor.submit(remoteConnect,tcp_host,tcp_port,connection_key,'https://mqtt.eu-latest.cumulocity.com' )
```

Comment:
The url cannot be hardcoded.

The command `sudo tedge config get c8y.url` returns the url to be used (without the `https://` prefix though).

This `url` can also be found in `/etc/tedge/tedge.toml`.

Reply:
fixed with next commit